### DISCUSSION TITLE
'Drainage' object has no attribute 'soln'

### DISCUSSION THREAD

**Discussion by JahanzebTariq** [PMEAL/OpenPNM#2688] [Q&A]
Greetings @jgostick and @ma-sadeghi ... I have been runing into this issue of 'Drainage' object has no attribute 'soln'... @jgostick sir you suggested me the solution to extract and install code from the devb branch and I did so after removing the previously installed OpenPNM and installed the dev version with the latest update like this:
![image](https://user-images.githubusercontent.com/99289270/217854175-07a85231-4d55-4501-927f-c28cb115bc0e.png)
but I still got this issue mentioned in the title.. and after your suggestion of:
```
drn.settings['quantity'] = 'throat.invasion_pressure'
drn.soln = {'throat.invasion_pressure': drn['throat.invasion_pressure']}
```
but I am still having the issue.. 
![image](https://user-images.githubusercontent.com/99289270/217855257-6f3a09a6-9335-4445-8709-074b2779461c.png)
@jgostick please help me get this one through.. I have been doing every possibility to get around this issue but it does not seem to solve..I know that you have solved this issue at your end but please do correct me where I'm falling short.. I humbly request you to address this issue as this issue has got me stuck for over a month and my project dead line is passing by really fast and I really need to visualize these percolation paths in the Paraview... **PLEASE DO RESOLVE** 
thank you

**Comment by jgostick** [comment]
We've been developing OpenPNM and PoreSpy for more than 10 years, and have shared every line of code with the world, free of charge.  I like to think that many people have benefited and we've somehow made the world a better place in our small way.  Howevrer, in return for our efforts we have received no material benefits at all.  We don't get paid.  Nobody has contributed any substantial code.  I have not been invited on any all-expense paid trips to give short courses.  I can count on one hand how many people have even reached out to thank me.  Had we kept this code completely private we could have saved ourselves 1000's of hours of maintenance and dedicated this time to our research instead, publishing 2x more papers and working on other more fruitful projects (i.e. start-up company).  So when you make demands of us in ALL CAPS you are way out of line.  Perhaps there are some nuances being lost in your text message, and I do see that you use the word 'please' in your request so I appreciate your manners, but you are easily the most demanding person we've ever communicated with on Github.  What's especially grating is that your problems are all due to your lack of experience in this area, and you're trying to "outsource" that deficit on to me and my team, instead of spending a few weeks of your own time learning the python ecosystem. 

Now, about your problem: I suspected that the Jupyter kernal you are running is not seeing the new installation of OpenPNM. To confirm this I opened a "Binder" site which does a fresh install of the code in a cloud server, totally independent of my personal setup, and my suggested temporary fix worked.  

You can try it yourself here: https://github.com/jgostick/openpnm-binder.  There is a link to the Binder site at the bottom of the readme.  It will launch a cloud instance of openpnm inside a jupyter-lab interface.  Open notebook number 3 and run it.  At the end of the notebook, create a new cell and add:

```python
mip.settings['quantity'] = 'throat.invasion_pressure'
mip.soln = {'throat.invasion_pressure': mip['throat.invasion_pressure']}
op.io.project_to_vtk(pn.project)
```

As you can see from the screenshot below, this worked for me, and the 'vtp' file that is created appears in the file browser on the left.

![screen_capture](https://user-images.githubusercontent.com/5092049/217906230-10623e84-3450-4eae-a6be-a8a27b40110f.png)

I happen to be extremely busy right now with my "other" job duties, so I am particularly touchy about this subject, but I think its fair to say that you've used up your quota of 'free tech support' from me and my team.
